### PR TITLE
Does not close stdout when running DlmeJsonResourceWriter. Closes #54.

### LINKED ISSUE #54: Traject LineWriter closes stdout
Steps to reproduce:
```
docker run --rm -e SKIP_FETCH_CONFIG=true                 -e SKIP_FETCH_DATA=true                 -v $(pwd)/.:/opt/traject                 -v $(pwd)/../dlme-traject:/opt/traject/config                 -v $(pwd)/../dlme-metadata:/opt/traject/data                 -v $(pwd)/output:/opt/traject/output                 suldlss/dlme-transform:latest                 stanford/maps/data
```

Expect result:
All of the files under `stanford/maps/data` are transformed.

Actual result:
First file processes and then error.
```
/usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:30:in `write': closed stream (IOError)
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:30:in `puts'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:30:in `_write'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:41:in `block in put'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:40:in `synchronize'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/line_writer.rb:40:in `put'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/indexer.rb:558:in `block (3 levels) in process'
	from /usr/local/bundle/gems/traject-3.0.0/lib/traject/thread_pool.rb:123:in `block in maybe_in_thread_pool'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:348:in `run_task'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:337:in `block (3 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `loop'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:320:in `block (2 levels) in create_worker'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `catch'
	from /usr/local/bundle/gems/concurrent-ruby-1.1.4/lib/concurrent/executor/ruby_thread_pool_executor.rb:319:in `block in create_worker'
```

### PR DESCRIPTION
**Pull request by justinlittman**

### CODE REVIEW

File: lib/dlme_json_resource_writer.rb [hunk 1]
```diff
@@ -21,6 +21,10 @@ def serialize(context)
       "#{JSON.pretty_generate(adjusted)}"
   end
 
+  def close
+    @output_file.close unless (@output_file.nil? || @output_file.tty? || @output_file == $stdout)
```
Comment (jcoyne) [inline lib/dlme_json_resource_writer.rb]: Can you explain why we need to check for `tty?`

Comment (jcoyne) [reply lib/dlme_json_resource_writer.rb]: or nil for that matter?

Comment (jcoyne) [reply lib/dlme_json_resource_writer.rb]: Should this patch be made upstream? https://github.com/traject/traject/blob/master/lib/traject/line_writer.rb#L60

Comment (jcoyne) [reply lib/dlme_json_resource_writer.rb]: Can you refactor to 
```ruby
def close
  return if  @output_file == $stdout
  super
end
```

**Comment by justinlittman** [comment]
Upstream PR: https://github.com/traject/traject/pull/202

**Pull request merged**